### PR TITLE
api: support 5.0

### PR DESCRIPTION
**Pull request by setval**
for v2

### CODE REVIEW

**Comment by DiscoreMe** [comment]
https://github.com/tucnak/telebot/issues/404

File: webhook.go [hunk 1]
```diff
@@ -49,6 +49,9 @@ type Webhook struct {
 	ErrorUnixtime  int64  `json:"last_error_date"`
 	ErrorMessage   string `json:"last_error_message"`
 
+	IpAddress          string `json:"ip_address"`
```
Comment (demget) [inline webhook.go]: I believe simply `IP` is enough here.

File: webhook.go [hunk 2]
```diff
@@ -49,6 +49,9 @@ type Webhook struct {
...
+	DropPendingUpdates bool   `json:"drop_pending_updates"`
```
Comment (demget) [inline webhook.go]: As with the `IP` field it's better to simplify it to `DropUpdates`.

File: chat.go [hunk 3]
```diff
@@ -50,6 +51,11 @@ type Chat struct {
 	SlowMode         int        `json:"slow_mode_delay,omitempty"`
 	StickerSet       string     `json:"sticker_set_name,omitempty"`
 	CanSetStickerSet bool       `json:"can_set_sticker_set,omitempty"`
+	LinkedChatID     int64      `json:"linked_chat_id,omitempty"`
+	ChatLocation     *struct {
```
Comment (demget) [inline chat.go]: Let's make another struct for `ChatLocation`, down here in `chat.go`.

File: admin.go [hunk 4]
```diff
@@ -148,12 +148,16 @@ func (b *Bot) Ban(chat *Chat, member *ChatMember) error {
 }
 
 // Unban will unban user from chat, who would have thought eh?
-func (b *Bot) Unban(chat *Chat, user *User) error {
+func (b *Bot) Unban(chat *Chat, user *User, isBanned ...bool) error {
```
Comment (demget) [inline admin.go]: `isBanned` is not entirely correct here, I'd name it `forBanned` which clearly shows that with this parameter, the function is going to affect only banned users.

File: telebot.go [hunk 5]
```diff
@@ -144,6 +144,11 @@ const (
 	//
 	// Handler: func(*Message)
 	OnVoiceChatPartecipantsInvited = "\avoice_chat_partecipants_invited"
+
+	// Will fire on ProximityAlertTriggered
+	//
+	// Handler: func(*Message)
+	OnProximityAlertTriggered = "\aproximity_alert_triggered"
```
Comment (demget) [inline telebot.go]: My proposal is to rename it to `OnProximityAlert` and similarly for other related structs or fields, as word _triggered_ is already implied when you handle events.

File: bot.go [hunk 6]
```diff
@@ -1395,15 +1395,32 @@ func (b *Bot) Pin(msg Editable, options ...interface{}) error {
...
 	}
+	if len(messageID) > 0 {
+		params["message_id"] = strconv.Itoa(messageID[0])
+	}
 
 	_, err := b.Raw("unpinChatMessage", params)
 	return err
 }
 
+// Unpin unpins all messages in a supergroup or a channel.
```
Comment (demget) [inline bot.go]: Documentation for the function copy-pasted, but wasn't edited for `UnpinAll`.

File: media.go [hunk 7]
```diff
@@ -195,9 +195,24 @@ type Location struct {
 	// Longitude
 	Lng float32 `json:"longitude"`
 
+	// Horizontal Accuracy
+	HA *float32 `json:"horizontal_accuracy,omitempty"`
```
Comment (demget) [inline media.go]: In this particular case, it's honestly better not to abbreviate words, _horizontal accuracy_ isn't something that can be recognized by these two letters.

File: media.go [hunk 8]
```diff
@@ -195,9 +195,24 @@ type Location struct {
...
+	Heading int `json:"heading,omitempty"`
+
 	// Period in seconds for which the location will be updated
 	// (see Live Locations, should be between 60 and 86400.)
 	LivePeriod int `json:"live_period,omitempty"`
+
+	ProximityAlertRadius int `json:"proximity_alert_radius,omitempty"`
+}
+
+// ProximityAlertTriggered object represents sent whenever
```
Comment (demget) [inline media.go]: I suppose "object represents" words should be removed here.

File: chat.go [hunk 9]
```diff
@@ -78,9 +78,10 @@ func (c *Chat) Recipient() string {
 type ChatMember struct {
 	Rights
 
-	User  *User        `json:"user"`
-	Role  MemberStatus `json:"status"`
-	Title string       `json:"custom_title"`
+	User        *User        `json:"user"`
+	Role        MemberStatus `json:"status"`
+	Title       string       `json:"custom_title"`
+	IsAnonymous bool         `json:"is_anonymous"`
```
Comment (demget) [inline chat.go]: As we named `is_anonymous` field `Anonymous` once, we should use this name for further similar fields to preserve consistency. https://github.com/tucnak/telebot/blob/62cfa356f7f6a2c9e0ad011c32f3df2c076a57b4/polls.go#L22

**Review by demget — CHANGES_REQUESTED** [review]
Some small adjustments to be done, everything else is fine.

**Review by demget — APPROVED** [review]

**Comment by demget** [comment]
Such a great improvement for Telebot! Thank you for your work!

**Pull request merged**